nilearn/glm/tests/test_first_level.py

```diff
@@ -2710,3 +2712,60 @@ def test_first_level_from_bids_surface(tmp_path):
     )
 
     _check_output_first_level_from_bids(n_sub, models, imgs, events, confounds)
+
+
+@pytest.mark.slow
+def test_generate_report(shape_4d_default):
+    """Test nilearn.glm.first_level.FirstLevelModel.generate_report for
+    warnings depending on threshold and height_control values.
+    """
+    mask, fmri_data, design_matrices = generate_fake_fmri_data_and_design(
+        shapes=[shape_4d_default], rk=3
+    )
+
+    flm = FirstLevelModel(mask_img=mask).fit(
+        fmri_data[0], design_matrices=design_matrices[0]
+    )
+
+    contrasts = [
+        np.asarray([1, 0, 0]),
+        np.asarray([1, 1, 0]),
+        np.asarray([1, 1, 1]),
+    ]
+
+    # should return a single warning, set threshold to None
+    with pytest.warns(UserWarning, match="'threshold' to None"):
+        flm.generate_report(contrasts=contrasts)
+
+    # should return a single warning, set height_control to None
+    with pytest.warns(UserWarning, match="'height_control' to None"):
+        flm.generate_report(contrasts=contrasts, threshold=3)
+
+    # should return a single warning default threshold deprecation
+    with warnings.catch_warnings(record=True) as warning_list:
+        flm.generate_report(contrasts=contrasts, height_control=None)
+        n_warnings = len(
+            [x for x in warning_list if issubclass(x.category, FutureWarning)]
+        )
+        assert n_warnings == 1
+
+    # should return a single warning, set height_control to None
+    with pytest.warns(UserWarning, match="'height_control' to None"):
+        flm.generate_report(contrasts=contrasts, threshold=DEFAULT_Z_THRESHOLD)
+
+    # no warning expected
+    with warnings.catch_warnings(record=True) as warning_list:
+        flm.generate_report(
+            contrasts=contrasts,
+            height_control=None,
+            threshold=DEFAULT_Z_THRESHOLD,
+        )
```

Comment:
I guess you cannot just do ``assert len(wartning_list) == 0`` because some other warnings are thrown, right?